### PR TITLE
Fenicsx update dependency and build for main branch

### PR DESCRIPTION
**Pull request by chrisrichardson**
Minor fixes

### CODE REVIEW

**Review by garth-wells — DISMISSED** [review]

**Review by mscroggs — DISMISSED** [review]

File: var/spack/repos/builtin/packages/fenics-basix/package.py [hunk 1]
```diff
@@ -23,3 +23,6 @@ class FenicsBasix(CMakePackage):
     depends_on("xtl@0.7.2:")
     depends_on("xtensor@0.23.10:")
     depends_on("blas", type=("build", "run"))
+
+    with when("@main"):
+        root_cmakelists_dir = "cpp"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fenics-basix/package.py]: The `when` context manager works only with directives. Does this have any effect for you?

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/fenics-basix/package.py (hunk 1 above)
Comment (chrisrichardson) [reply var/spack/repos/builtin/packages/fenics-basix/package.py]: @alalazo - I am not sure. It seems to work OK as it is. I was not sure how else to set a variable, which depends on the version.

Comment (alalazo) [reply var/spack/repos/builtin/packages/fenics-basix/package.py]: I guess it works also without the `when`. Did you try both `main` and other versions?

> I was not sure how else to set a variable, which depends on the version.

In Python you can code a "property":
```python
@property
def root_cmake_lists_dir(self):
    if self.spec.satisfies("@main"):
        return "cpp"
    return None
```

Comment (chrisrichardson) [reply var/spack/repos/builtin/packages/fenics-basix/package.py]: Thanks - I have updated to use this.

Comment (garth-wells) [reply var/spack/repos/builtin/packages/fenics-basix/package.py]: @alalazo is anything more needed for the PR?

**Review by garth-wells — APPROVED** [review]

**Review by mscroggs — APPROVED** [review]

**Comment by chrisrichardson** [comment]
@alalazo - can this now be merged?

**Review by tgamblin — APPROVED** [review]

**Pull request merged**